scipy/spatial/transform/_rotation.pyx

```diff
@@ -1417,8 +1556,10 @@ cdef class Rotation:
             return np.asarray(rotvec)
 
     @cython.embedsignature(True)
-    def as_euler(self, seq, degrees=False):
+    def as_euler_from_matrix(self, seq, degrees=False):
```

Comment:
This method is not necessary in our updated approach to migrate to the new method. Right?

Reply:
I decided to leave it there to make it easier to test and compare both functions.
This way, both of them are accessed and executed in the same way! What do you think?

Reply:
Do you suggest using it for testing then? I thought we've agreed that comparison with ground truth angles is sufficient to adopt the new algorithm (at least from my point of view).

Reply:
I already implemented tests for testing both methods, but I can completely remove the old method (both `as_euler_from_matrix` and `_compute_euler_from_matrix`) if you prefer!

Reply:
Ok sorry, now I remember: testing the angle differences and angle sums is a bit awkward to implement, specially for asymetric cases (for a `parity` related problem). 
This is why I decided to leave the old method and compare: not only it's simpler (and then, safer), but I'm also comparing it to something we already know is right!

Reply:
Good, it's a practical decision.

In this case I suggest to hide the method from public API by naming it as `_as_euler_from_matrix`.

Also, as they share the preparation and verification code let's factor it out as another private method, if you don't mind.

Reply:
Alright, I'll leave it as a private method then!

> Also, as they share the preparation and verification code let's factor it out as another private method, if you don't mind.
`as_euler_from_matrix` is the verification/preparation code already, and `_compute_euler_from_matrix` already has an underscore, so I imagine it is already private?

Reply:
I mean that now `as_euler` and `_as_euler_from_matrix` have identical code before calling the actual conversion. 

I think it's reasonable to move this code into a separate method and call it from both methods. Although not critically necessary.

Reply:
I see! In this case I can create a new private method, maybe called `_prepare_axis_for_euler` that does this and calls either `_compute_euler_from_matrix` or `_compute_euler_from_quat` according to a parameter `algorithm`.

Then, both `as_euler` and `_as_euler_from_quat` will just pass their arguments with the desired extra `algorithm` parameter.

Alternatively, leaving the functions as they are at the moment has the advantage that when the decision to completely remove the older algorithm is made in the future, it will be simpler to just delete it.

What do you think?

Reply:
Yes, a generalizing method looks like a good solution here. Should reduce total number of lines.

It's unlikely we will remove from matrix conversion if it is used for testing.